app/components/transUnit/TransUnitService.js

```diff
@@ -170,7 +198,26 @@
        updateSaveButton);
 
     function modifyTranslationText(phrase, newText) {
-      phrase.newTranslation = newText;
+      if (phrase.plural) {
+        var transUnitCtrl = controllerList[phrase.id];
+        phrase.newTranslations[transUnitCtrl.focusedTranslationIndex] = newText;
+      } else {
+        phrase.newTranslation = newText;
+      }
+      EventService.emitEvent(EventService.EVENT.TRANSLATION_TEXT_MODIFIED,
+        phrase);
+      EventService.emitEvent(EventService.EVENT.FOCUS_TRANSLATION,
+        phrase);
+    }
+
+    function modifyTranslationsText(phrase, newTexts) {
```

Comment:
I'm sure we could simplify the code a lot by just treating everything the same whether it is plural or not. For anything that is not plural, just treat it as thought it is a plural with 1 source and 1 translation. There are lots of places in the code that we branch the behaviour, but there are only a few places we should need to treat them any different: receiving data from the server (populating the source/target arrays from either `content` or `contents`) and sending the data back to the server (putting the data in either `content` or `contents` have to be used correctly), and maybe a slightly different display template.

When can we simplify this code? I suspect there is insufficient time this sprint to make that much of a change.